Modules/_ctypes/cfield.c

```diff
@@ -87,25 +99,22 @@ PyCField_FromDesc(PyObject *desc, Py_ssize_t index,
     } else if (bitsize /* this is a bitfield request */
         && *pfield_size /* we have a bitfield open */
         && dict->size * 8 >= *pfield_size
-        && (*pbitofs + bitsize) <= dict->size * 8) {
+        && (((*pbitofs + bitsize) <= dict->size * 8) || pack)) {
```

Comment:
When we have a packed bitfield and it doesn't fit the current open bitfield, always expand it.

Reply:
I'm quite convinced this comment should go in the source code.